api-server/src/server/boot/authentication.js

```diff
@@ -188,3 +194,40 @@ function createGetPasswordlessAuth(app) {
     );
   };
 }
+
+function mobileLogin(app) {
+  const {
+    models: { User }
+  } = app;
+  return async function getPasswordlessAuth(req, res, next) {
+    const {
+      headers: { token: accessToken }
+    } = req;
+
+    const auth0Res = await fetch(
+      `https://${process.env.AUTH0_DOMAIN}/userinfo`,
+      {
+        headers: { Authorization: `Bearer ${accessToken}` }
+      }
+    );
+
+    const { email } = await auth0Res.json();
+
+    if (!isEmail(email)) {
+      return next(
+        wrapHandledError(new TypeError('decoded email is invalid'), {
+          type: 'info',
+          message: 'The email encoded in the link is incorrectly formatted',
+          redirectTo: `${origin}/signin`
```

Comment:
What's the idea behind this redirect? Mobile users can't (or shouldn't?) be trying to use the web signin, so why redirect there?

Reply:
Good catch. There won't be a redirect, we'll be directly stopping the flow there.